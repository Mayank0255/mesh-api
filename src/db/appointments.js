import { performQuery } from ".";

const getAppointmentQuery = `SELECT
	*
FROM
	appointments
WHERE
	id = $1`;

const createAppointmentQuery = `INSERT INTO appointments (type, date, notes, member_id, building_id, request_id, acuity_id)
	VALUES($1, $2, $3, $4, $5, $6, $7)
RETURNING
	*`;

export async function getAppointment(id) {
	return performQuery(getAppointmentQuery, [id]);
}

export async function createAppointment(appointment) {
	return performQuery(createAppointmentQuery, [
		appointment.type,
		appointment.date,
		appointment.notes,
		appointment.member_id,
		appointment.building_id,
		appointment.request_id,
		appointment.acuity_id
	]);
}
