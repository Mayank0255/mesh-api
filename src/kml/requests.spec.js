import { performQuery } from '../db';
jest.mock('../db');
import { getRequestsKML } from './requests';

describe('getRequestsKML', () => {
  it('renders KML for current install requests', async () => {
    performQuery.mockResolvedValue([
      {
        id: 7059,
        status: 'open',
        apartment: null,
        roof_access: true,
        date: new Date('2020-09-18T12:42:20.811Z'),
        osticket_id: null,
        member_id: 6087,
        building_id: 5758,
        building: {
          id: 5758,
          address: 'Redacted',
          lat: 40.6604126,
          lng: -73.9428339,
          alt: 32,
          bin: 3114902,
          notes: null
        },
        panoramas: null
      },
      {
        id: 7060,
        status: 'open',
        apartment: null,
        roof_access: false,
        date: new Date('2020-10-18T12:42:20.811Z'),
        osticket_id: null,
        member_id: 6088,
        building_id: 5759,
        building: {
          id: 5759,
          address: 'Redacted',
          lat: 41.6604126,
          lng: -72.9428339,
          alt: 33,
          bin: 3114911,
          notes: null
        },
        panoramas: null
      },
    ]);

    expect(await getRequestsKML()).toMatchSnapshot()
  });
});
