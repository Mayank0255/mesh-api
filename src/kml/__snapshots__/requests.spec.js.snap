// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRequestsKML renders KML for current install requests 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<kml xmlns=\\"http://www.opengis.net/kml/2.2\\">
    <Document>
        <Style id=\\"request\\">
    <IconStyle>
        <scale>0.5</scale> 
        <Icon>
            <href>https://i.imgur.com/oVFMyJU.png</href>
        </Icon>
        <hotSpot xunits=\\"fraction\\" yunits=\\"fraction\\" x=\\"0.5\\" y=\\"0.5\\"></hotSpot>
    </IconStyle>
    <LabelStyle>
        <scale>0</scale>
    </LabelStyle>
</Style>,<Style id=\\"panoRequest\\">
    <IconStyle>
        <scale>0.5</scale> 
        <Icon>
            <href>https://i.imgur.com/uj6HMxZ.png</href>
        </Icon>
        <hotSpot xunits=\\"fraction\\" yunits=\\"fraction\\" x=\\"0.5\\" y=\\"0.5\\"></hotSpot>
    </IconStyle>
    <LabelStyle>
        <scale>0</scale>
    </LabelStyle>
</Style>,
<Placemark>
	<name>7059</name>
	<ExtendedData>
		<Data name=\\"ID\\">
    <value>7059</value>
</Data>
		<Data name=\\"Date\\">
    <value>Fri Sep 18 2020</value>
</Data>
		<Data name=\\"Roof\\">
    <value>Yes</value>
</Data>
		<Data name=\\"Links\\">
    <value><a href=\\"https://dashboard.nycmesh.net/requests/7059\\" style=\\"margin-right: 1rem;\\">Dashboard →</a> <a href=\\"https://support.nycmesh.net/scp/tickets.php?a=search&amp;query=7059\\" style=\\"margin-right: 1rem;\\">Tickets →</a></value>
</Data>
		<Data name=\\"Panos\\">
    <value></value>
</Data>
	</ExtendedData>
	<Point>
		<altitudeMode>absolute</altitudeMode>
		<coordinates>-73.9428339,40.6604126,32</coordinates>
	</Point>
	<styleUrl>#request</styleUrl>
</Placemark>,
<Placemark>
	<name>7060</name>
	<ExtendedData>
		<Data name=\\"ID\\">
    <value>7060</value>
</Data>
		<Data name=\\"Date\\">
    <value>Sun Oct 18 2020</value>
</Data>
		<Data name=\\"Roof\\">
    <value>No</value>
</Data>
		<Data name=\\"Links\\">
    <value><a href=\\"https://dashboard.nycmesh.net/requests/7060\\" style=\\"margin-right: 1rem;\\">Dashboard →</a> <a href=\\"https://support.nycmesh.net/scp/tickets.php?a=search&amp;query=7060\\" style=\\"margin-right: 1rem;\\">Tickets →</a></value>
</Data>
		<Data name=\\"Panos\\">
    <value></value>
</Data>
	</ExtendedData>
	<Point>
		<altitudeMode>absolute</altitudeMode>
		<coordinates>-72.9428339,41.6604126,33</coordinates>
	</Point>
	<styleUrl>#request</styleUrl>
</Placemark>
    </Document>
</kml>"
`;
